services/drupal/web/modules/custom/view_mode_select_by_token/src/Plugin/Field/FieldFormatter/RenderedWithSelect.php

```diff
@@ -0,0 +1,97 @@
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+//    $this->viewMode = \Drupal::token()->replace($this->getSetting('view_mode'));
+    // @Todo dependency inject this
+    $entity = $items->getEntity();
+    $entity_type = $entity->getEntityTypeId();
+    $view_mode = \Drupal::token()->replace($this->getSetting('view_mode'), [$entity_type => $entity]);
+    $elements = [];
+
+    foreach ($this->getEntitiesToView($items, $langcode) as $delta => $entity) {
+      // Due to render caching and delayed calls, the viewElements() method
+      // will be called later in the rendering process through a '#pre_render'
+      // callback, so we need to generate a counter that takes into account
+      // all the relevant information about this field and the referenced
+      // entity that is being rendered.
+      $recursive_render_id = $items->getFieldDefinition()->getTargetEntityTypeId()
+        . $items->getFieldDefinition()->getTargetBundle()
+        . $items->getName()
+        // We include the referencing entity, so we can render default images
+        // without hitting recursive protections.
+        . $items->getEntity()->id()
+        . $entity->getEntityTypeId()
+        . $entity->id();
+
+      if (isset(static::$recursiveRenderDepth[$recursive_render_id])) {
+        static::$recursiveRenderDepth[$recursive_render_id]++;
+      }
+      else {
+        static::$recursiveRenderDepth[$recursive_render_id] = 1;
+      }
+
+      // Protect ourselves from recursive rendering.
+      if (static::$recursiveRenderDepth[$recursive_render_id] > static::RECURSIVE_RENDER_LIMIT) {
+        $this->loggerFactory->get('entity')->error('Recursive rendering detected when rendering entity %entity_type: %entity_id, using the %field_name field on the %parent_entity_type:%parent_bundle %parent_entity_id entity. Aborting rendering.', [
+          '%entity_type' => $entity->getEntityTypeId(),
+          '%entity_id' => $entity->id(),
+          '%field_name' => $items->getName(),
+          '%parent_entity_type' => $items->getFieldDefinition()->getTargetEntityTypeId(),
+          '%parent_bundle' => $items->getFieldDefinition()->getTargetBundle(),
+          '%parent_entity_id' => $items->getEntity()->id(),
+        ]);
+        return $elements;
+      }
+
+      $view_builder = $this->entityTypeManager->getViewBuilder($entity->getEntityTypeId());
+      $elements[$delta] = $view_builder->view($entity, $view_mode, $entity->language()->getId());
+
+      // Add a resource attribute to set the mapping property's value to the
+      // entity's url. Since we don't know what the markup of the entity will
+      // be, we shouldn't rely on it for structured data such as RDFa.
+      if (!empty($items[$delta]->_attributes) && !$entity->isNew() && $entity->hasLinkTemplate('canonical')) {
+        $items[$delta]->_attributes += ['resource' => $entity->toUrl()->toString()];
+      }
+    }
+
+    return $elements;
```

Comment:
Basically all of this can be discarded; it's just a copy from the parent. You should be able to just set the view mode setting then call the parent method:
```suggestion
$this->setSetting('view_mode', $view_mode);
return parent::viewElements($items, $langcode);
```